src/components/MessageCard/MessageCard.jsx

```diff
@@ -53,11 +54,17 @@ export const MessageCard = React.memo(
 
       const hasImage = useCallback(() => image && image.url, [image])
 
+      const makeMessageVisible = useCallback(() => {
+        setTimeout(() => {
+          isMounted.current && setVisible(true)
+        }, 0)
+      }, [])
+
       useEffect(() => {
-        if (inProp && !hasImage() && !isShown.current) {
+        if (inProp && !hasImage() && !isShown.current && !video) {
           makeMessageVisible()
         }
-      }, [inProp, hasImage])
+      }, [inProp, hasImage, video, makeMessageVisible])
```

Comment:
Since video is on object (and I assume there is a reason for that) and `useEffect` compares values referencially and not deeply, it might trigger rerendering when you don't want to.

You can either use the `use-deep-compare-effect` package (already installed, works the same as useEffect) or if you know that `video.html` is always a string, extract that and pass it to useEffect:

```js
const { html } = video

useEffect(() => {
  if (inProp && !hasImage() && !isShown.current && !video) {
    makeMessageVisible()
  }
}, [inProp, hasImage, html, makeMessageVisible])
```

Reply:
Great catch! I'll use comparing the `html` :)